### ISSUE TITLE
Allow report analysis of specific java files

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1632]
**version**

 master

**platform**

JVM

**problem**
It'd help clojure-lsp tests if when we lint a `java` file directly, we receive `java-class-definition` analysis, ATM clj-kondo lint as a clojure file it seems

```java
package clojure_sample;

class Foo {
    private String myField;

}
```

```
clj-kondo --lint src-java/clojure_sample/Foo.java --config '{:output {:format :json :analysis {:java-class-definitions true :java-class-usages true} :canonical-paths true}}' | jq .analysis
```

OUTPUT
```json
{
  "namespace-definitions": [],
  "namespace-usages": [],
  "var-definitions": [],
  "var-usages": [
    {
      "end-row": 1,
      "name-end-col": 8,
      "name-end-row": 1,
      "name-row": 1,
      "name": "package",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 1,
      "name-col": 1,
      "end-col": 8,
      "row": 1,
      "to": "clj-kondo/unknown-namespace"
    },
    {
      "end-row": 1,
      "name-end-col": 23,
      "name-end-row": 1,
      "name-row": 1,
      "name": "clojure_sample",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 9,
      "name-col": 9,
      "end-col": 23,
      "row": 1,
      "to": "clj-kondo/unknown-namespace"
    },
    {
      "fixed-arities": [
        1
      ],
      "end-row": 3,
      "name-end-col": 6,
      "name-end-row": 3,
      "name-row": 3,
      "name": "class",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 1,
      "name-col": 1,
      "end-col": 6,
      "row": 3,
      "to": "clojure.core"
    },
    {
      "end-row": 3,
      "name-end-col": 10,
      "name-end-row": 3,
      "name-row": 3,
      "name": "Foo",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 7,
      "name-col": 7,
      "end-col": 10,
      "row": 3,
      "to": "clj-kondo/unknown-namespace"
    },
    {
      "end-row": 4,
      "name-end-col": 12,
      "name-end-row": 4,
      "name-row": 4,
      "name": "private",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 5,
      "name-col": 5,
      "end-col": 12,
      "row": 4,
      "to": "clj-kondo/unknown-namespace"
    },
    {
      "end-row": 4,
      "name-end-col": 27,
      "name-end-row": 4,
      "name-row": 4,
      "name": "myField",
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "from": "user",
      "col": 20,
      "name-col": 20,
      "end-col": 27,
      "row": 4,
      "to": "clj-kondo/unknown-namespace"
    }
  ],
  "java-class-definitions": [],
  "java-class-usages": [
    {
      "end-row": 4,
      "name-end-col": 19,
      "name-end-row": 4,
      "name-row": 4,
      "filename": "/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "col": 13,
      "class": "java.lang.String",
      "name-col": 13,
      "uri": "file:/home/greg/dev/clojure-sample/src-java/clojure_sample/Foo.java",
      "end-col": 19,
      "row": 4
    }
  ]
}

```
